module/admin/helper.js

```diff
@@ -128,4 +140,455 @@ module.exports = class AdminHelper {
+					)
+					// Delete all associated entities tied to projectTemplateIds
+					if (programRoleMappingId.nModified > 0) {
+						pullProgramFromUserExtensionCount = programRoleMappingId.nModified
+					}
+
+					// Delete all associated resources for collected projectTemplateIds
+					if (projectTemplateIds.length > 0) {
+						const result = await this.deleteAssociatedResources(
+							projectTemplateIds,
+							tenantId,
+							orgId,
+							deletedBy
+						)
+
+						if (result.success) {
+							projectTemplateDeletedCount += result.result.projectTemplateDeletedCount
+							certificateTemplateDeletedCount += result.result.certificateTemplateDeletedCount
+							taskDeletedCount += result.result.taskDeletedCount
+							surveyCount += result.result.surveyCount
+							surveySubmissionCount += result.result.surveySubmissionCount
+							observationCount += result.result.observationCount
+							observationSubmissionCount += result.result.observationSubmissionCount
+						}
+					}
+
+					// Delete the program itself
+					await programsQueries.deletePrograms(programFilter)
+					programDeletedCount++
+					// Publish Kafka event
+					// {
+					// 	"topic": "RESOURCE_DELETION_TOPIC",
+					// 	"messages": "{\"entity\":\"resource\",\"type\":\"program\",\"eventType\":\"delete\",\"entityId\":\"682c1526ba875600144d93bc\",\"deleted_By\":1,\"tenant_code\":\"shikshagraha\",\"organization_id\":[\"blr\"]}"
+					//   }
+					await this.pushResourceDeleteKafkaEvent(
+						CONSTANTS.common.PROGRAM,
+						resourceId,
+						deletedBy,
+						tenantId,
+						orgId
+					)
+
+					// Add deletion entry to logs
+					await this.addDeletionLog(resourceIdsWithType, deletedBy)
+					return resolve({
+						success: true,
+						message: CONSTANTS.apiResponses.PROGRAM_RESOURCE_DELETED,
+						result: {
+							programDeletedCount,
+							solutionDeletedCount,
+							projectTemplateDeletedCount,
+							certificateTemplateDeletedCount,
+							taskDeletedCount,
+							surveyCount,
+							surveySubmissionCount,
+							observationCount,
+							observationSubmissionCount,
+							pullProgramFromUserExtensionCount,
+							projectDeletedCount,
+						},
+					})
+				} else if (resourceType === CONSTANTS.common.SOLUTION) {
+					// Handle deletion for a solution
+					const solutionFilter = {
+						_id: resourceId,
+						tenantId,
+					}
+					// Fetch solution document
+					const solutionDetails = await solutionsQueries.solutionsDocument(solutionFilter, [
+						'projectTemplateId',
+					])
+					if (!solutionDetails || !solutionDetails.length) {
+						throw {
+							status: HTTP_STATUS_CODE.bad_request.status,
+							message: CONSTANTS.apiResponses.SOLUTION_NOT_FOUND,
+						}
+					}
+					// Delete projects linked to the solution
+					let projecFilter = {
+						solutionId: { $in: resourceId },
+					}
+					let deletedProjectIds = await projectQueries.deleteProjects(projecFilter)
+
+					if (deletedProjectIds.deletedCount > 0) {
```

Comment:
@MallanagoudaB what is the issue in assigning deletedCount without the if condition?